src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -101,11 +103,11 @@ runMiner v mr = do
         pure $ localTest gen miners
 
     powMiner :: IO (BlockHeader -> IO BlockHeader)
-    powMiner = case _configRemoteMiners conf of
-        [] -> pure $ localPOW v
-        rs -> do
+    powMiner = case NEL.nonEmpty $ _configRemoteMiners conf of
```

Comment:
Why are we doing this?  It doesn't look like we gain anything here because we're still casing.  The fact that we have to add the partition function below seems to make this a net loss in my mind.

Reply:
It's convenient (in terms of type safety) to have the following signature on the other end of this:
```haskell
remoteMining :: Manager -> NESet BaseUrl -> BlockHeader -> IO BlockHeader
```
It means we won't be wasting calls on potentially duplicate endpoints. And since `powMiner` has the signature:
```haskell
    powMiner :: IO (BlockHeader -> IO BlockHeader)
```
these container conversions are only performed once.

Reply:
Give me a moment. I just improved this, and I'm going to improve it further.

Reply:
Give me a moment, I just improved this, and I'm going to improve it further.